gui/src/app/Settings/UserSettingsProvider.tsx

```diff
@@ -0,0 +1,155 @@
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import CssBaseline from "@mui/material/CssBaseline";
+import useMediaQuery from "@mui/material/useMediaQuery";
+
+import {
+  FunctionComponent,
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+import { useDialogControls } from "@SpComponents/CloseableDialog";
+
+import { SettingsTab } from "./SettingsWindow";
+import {
+  publicCompilationServerUrl,
+  ThemeSetting,
+  UserSettingsContext,
+} from "./UserSettings";
+
+const colorPalette = {
+  primary: {
+    main: "#E16D44",
+  },
+  secondary: {
+    main: "#44B8E1",
+  },
+} as const;
+
+const defaultSettings = JSON.parse(
```

Comment:
It occurs to me that it'd be clearer to call this "savedSettings" or "storedSettings" or something--as they aren't defaults, they're what we've saved from the user's explicitly stated preferences.